modules/kafka/src/main/java/com/devonfw/module/kafka/common/messaging/trace/impl/MessageSpanInjector.java

```diff
@@ -0,0 +1,44 @@
+package com.devonfw.module.kafka.common.messaging.trace.impl;
+
+import static brave.internal.HexCodec.toLowerHex;
+
+import java.util.Optional;
+import java.util.UUID;
+
+import org.springframework.util.ObjectUtils;
+
+import com.devonfw.module.kafka.common.messaging.impl.MessageKafkaPayloadBuilder;
+import com.devonfw.module.logging.common.api.DiagnosticContextFacade;
+import com.devonfw.module.logging.common.impl.DiagnosticContextFacadeImpl;
+
+import brave.propagation.TraceContext;
+
+/**
+ * @author ravicm
+ *
+ */
+public class MessageSpanInjector implements TraceContext.Injector<MessageKafkaPayloadBuilder<?>> {
```

Comment:
Devon does not use b3 headers currently but a simple correlation ID (see com.devonfw.module.logging.common.api.LoggingConstants). Please align the implementation with that.

Reply:
for this am awaiting for the decision to remove the Tracer implementation or not .

Reply:
we have discussed in last architect meeting to keep tracer implementation.